src/cascadia/WindowsTerminal/NonClientIslandWindow.cpp

```diff
@@ -38,162 +36,75 @@ NonClientIslandWindow::~NonClientIslandWindow()
     // If we're maximized, we don't want to use the frame as our margins,
     // instead we want to use the margins from the maximization. If we included
     // the left&right sides of the frame in this calculation while maximized,
     // you' have a few pixels of the window border on the sides while maximized,
     // which most apps do not have.
     if (_isMaximized)
     {
-        clientWidth -= (_maximizedMargins.cxLeftWidth + _maximizedMargins.cxRightWidth);
-        clientHeight -= (margins.cyTopHeight + _maximizedMargins.cyBottomHeight);
-        clientOrigin.X = static_cast<short>(_maximizedMargins.cxLeftWidth);
+        windowsWidth -= (_maximizedMargins.cxLeftWidth + _maximizedMargins.cxRightWidth);
+        windowsHeight -= (_maximizedMargins.cyBottomHeight + _maximizedMargins.cyTopHeight);
     }
     else
     {
-        // Remove the left and right width of the frame from the client area
-        clientWidth -= (margins.cxLeftWidth + margins.cxRightWidth);
-        clientHeight -= (margins.cyTopHeight + margins.cyBottomHeight);
+        windowsWidth -= dragX * 2;
+        windowsHeight -= dragY * 2;
     }
 
-    // The top maximization margin is already included in the GetFrameMargins
-    // calcualtion.
-
-    return Viewport::FromDimensions(clientOrigin,
-                                    { static_cast<short>(clientWidth), static_cast<short>(clientHeight) });
-}
-
-// Method Description:
-// - called when the size of the window changes for any reason. Updates the
-//   sizes of our child Xaml Islands to match our new sizing.
-void NonClientIslandWindow::OnSize()
-{
-    auto clientArea = GetClientContentArea();
-    auto titlebarArea = GetTitlebarContentArea();
-
-    // update the interop window size
-    SetWindowPos(_interopWindowHandle, 0,
-                 clientArea.Left(),
-                 clientArea.Top(),
-                 clientArea.Width(),
-                 clientArea.Height(),
-                 SWP_SHOWWINDOW);
-
     if (_rootGrid)
     {
-        const SIZE physicalSize{ clientArea.Width(), clientArea.Height() };
-        const auto logicalSize = GetLogicalSize(physicalSize);
-        _rootGrid.Width(logicalSize.Width);
-        _rootGrid.Height(logicalSize.Height);
+        _rootGrid.Height((windowsHeight / scale) + 0.5);
+        _rootGrid.Width((windowsWidth / scale) + 0.5);
+    }
+
+    auto xPos = dragX;
+    auto yPos = dragY;
+    if (_isMaximized)
+    {
+        xPos = _maximizedMargins.cxLeftWidth;
+        yPos = _maximizedMargins.cyTopHeight;
     }
 
-    // update the interop window size
-    SetWindowPos(_nonClientInteropWindowHandle, 0,
-                 titlebarArea.Left(),
-                 titlebarArea.Top(),
-                 titlebarArea.Width(),
-                 titlebarArea.Height(),
-                 SWP_SHOWWINDOW);
+    SetWindowPos(_interopWindowHandle, NULL, xPos, yPos, windowsWidth, windowsHeight, SWP_SHOWWINDOW);
+
+    HRGN region = CreateRectRgn(0, 0, 0, 0);

```

Comment:
There's no error checking in this region block. `CreateRectRgn` can fail with NULL. `CombineRgn` can have a failure code. And `SetWindowRgn` is stated to be technically able to fail too.